File: src/prime.rs
use std::time::Instant;

fn is_prime(n: u32) -> bool {
    if n <= 1 {
        return false;
    }

    let end = (n as f32).sqrt() as u32;
    for i in 2..=end {
        if n % i == 0 {
            return false;
        }
    }
    true
}

fn main() {
    let start = Instant::now();

    let mut c= 0;
    for i in 0..9000000{
        if is_prime(i){
            c+=1;
        }
    }

    let elapsed = start.elapsed().as_millis();

    println!("{c}");
    println!("{elapsed}ms" );
}

File: src/prime.php
<?php

function is_prime($n) {
    if ($n <= 1) {
        return false;
    }
    
    $end = intval(sqrt($n));
    for ($i = 2; $i <= $end; $i++) {
        if ($n % $i == 0) {
            return false;
        }
    }
    return true;
}

$start = microtime(true) * 1000;
$c = 0;
for ($i = 0; $i < 9000000; $i++) {
    if (is_prime($i)) {
        $c++;
    }
}

echo "$c\n";
echo (microtime(true) * 1000 - $start);

?>
